File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.py
def solution(numbers, direction):
    answer = []  # 회전된 리스트를 저장할 변수

    if direction == "right":  # 오른쪽으로 회전하는 경우
        answer = numbers[-1:] + numbers[:-1]  # 마지막 원소를 앞으로 이동
    else:  # "left"인 경우 (왼쪽으로 회전)
        answer = numbers[1:] + numbers[:1]  # 첫 번째 원소를 뒤로 이동

    return answer  # 회전된 리스트 반환

# 테스트 예제 실행
print(solution([1, 2, 3], "right"))  # [3, 1, 2]
print(solution([4, 455, 6, 4, -1, 45, 6], "left"))  # [455, 6, 4, -1, 45, 6, 4]

File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.go
// 배열을 왼쪽 또는 오른쪽으로 한 칸씩 회전하는 Go 함수
func solution(numbers []int, direction string) []int {
	lenN := len(numbers)
	answer := make([]int, lenN)
	if direction == "right" {
		answer[0] = numbers[lenN-1]
		for i := 0; i < lenN-1; i++ {
			answer[i+1] = numbers[i]
		}
	} else if direction == "left" {
		for i := 1; i < lenN; i++ {
			answer[i-1] = numbers[i]
		}
		answer[lenN-1] = numbers[0]
	}
	return answer
}
